datalab/datalab_session/analysis/variable_star.py

```diff
@@ -29,25 +30,34 @@ def variable_star(input: dict, user: User):
 
   # Loop through each image's catalog and extract the target source's mag/magerr for the light curve
   for image in input.get("images"):
-    log.info(f"Processing image: {image.get('basename')} for variable star analysis...")
+    basename = image.get('basename')
 
     try:
-      file_path = FileCache().get_fits(image.get('basename'), input.get('source', 'archive'), user)
+      file_path = FileCache().get_fits(basename, input.get('source', 'archive'), user)
       cat_hdu = get_hdu(file_path, extension='CAT')
     except Exception as e:
-      log.error(f"Error retrieving catalog for image {image.get('basename')}: {e}")
+      log.error(f"Error retrieving catalog for image {basename}: {e}")
       continue
     
     target_source = find_target_source(cat_hdu, target_ra, target_dec)
 
     if target_source is None:
-      log.info(f"No matching source found for target coordinates: RA={target_ra}, DEC={target_dec} in image {image.get('basename')}")
-    else:
-      light_curve.append({
-        'mag': target_source['mag'],
-        'magerr': target_source['magerr'],
-        'observation_date': image.get("observation_date"),
-      })
+      log.info(f"No matching source found for target coordinates: RA={target_ra}, DEC={target_dec} in image {basename}")
+      continue
+
+    # Fallback calculating mag/magerr from flux/fluxerr if not in catalog columns
+    if(not 'mag' in target_source or not 'magerr' in target_source):
+      target_source['mag'], target_source['magerr'] = flux_to_mag(target_source['flux'], target_source['fluxerr'])
+
+    if target_source['mag'] is None or target_source['magerr'] is None:
+      log.warning(f"Invalid magnitude or magnitude error for target source in image {basename}. Skipping this source.")
```

Comment:
I think we should probably report back a dictionary of files that were excluded and their reason for exclusion, so the user can see that potentially in the UI and check their data or at least be aware of what was actually used to generate the result.